src/gretel_trainer/benchmark/custom/datasets.py

```diff
@@ -24,6 +25,8 @@ def __post_init__(self):
             rows, cols = self.data_source.shape
         self.row_count = rows
         self.column_count = cols
+        if self.public and not isinstance(self.data_source, str):
+            raise ValueError("datasets specified as dataframes can't be public")
```

Comment:
As noted above, I don't think there's a (public/documented/supported) way for a user to create a public dataset currently. That said, if we do add it, could we make this a `BenchmarkException` instead of `ValueError`?

Reply:
Done